### PR TITLE
[UI/UX] Toolbar page actions

### PR DESCRIPTION
**Pull request by andersonjeccel**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🔴 
| New feature/enhancement? (use the a.x branch)      | 🟢
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🔴 
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #... 

## Description

This PR groups related elements together to follow UX best practices.

When working on a details page, it makes sense to have a page header containing actions for the current resource (e.g. Edit action, to edit the current form, for example). 

Maybe not that much for a page that lists several resources for the same thing... In a first moment, you might look at the current state of our index pages and think that they look good.

image

but what if we add a widget to show a chart about your lead generation? what would be the "New" page action for, creating a contact or adding a new widget?

as we think about making the listing pages more attractive, is inevitable to consider adding content that creates a relationship with everything else you do in Mautic everyday

Placing the page actions on the toolbar specifically for listing pages aim to address this case, allowing us to keep developing content to make the tool get increasingly better without causing confusion to our users

image

---
### 📋 Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Open Contacts and have a look at the list toolbar, also interact
3. Repeat for all other listing pages

### CODE REVIEW

**Comment by andersonjeccel** [comment]
@abhisekmazumdar some changes to structure made e2e tests fail :)

**Comment by jos0405** [comment]
I would expect, that the live view works like a toggle. But actually I don't know if it's on.

**Comment by jos0405** [comment]
Perfect, couldn't as for more.
image

**Review by jos0405 — DISMISSED** [review]
Works really nice. Good change.

**Review by shinde-rahul — APPROVED** [review]
Code changes LGTM!!!

Thanks, @andersonjeccel, for this facelift. I loved the new look.

**Review by kuzmany — DISMISSED** [review]
Did basic tests, works for me 👍  Good job

**Review by RCheesley — DISMISSED** [review]
Tested this today and I think it's a really great improvement to the UX of the contextual search, but also just a nicer overall look for the contacts section.

**Review by shinde-rahul — APPROVED** [review]
Changes LGTM.

**Comment by abhisekmazumdar** [comment]
I feel the e2e `tests/acceptance/ContactManagementCest.php:importCSV` broken test is not relevant to these changes. I could be wrong; 
I tried running the `ddev composer e2e-test tests/acceptance/ContactManagementCest.php` for the 5.x branch too, which failed in the same way.

can someone confirm this?

**Comment by escopecz** [comment]
@abhisekmazumdar the e2e tests are flaky. We need someone to take the time to make them stable. 

@andersonjeccel there are some conflicts to resolve.

**Comment by abhisekmazumdar** [comment]
> @abhisekmazumdar the e2e tests are flaky. We need someone to take the time to make them stable.

@escopecz 
Do we have an issue created? where we could create a plan to identify where to start or what missing parts need improvement?
I don't have much idea about these test. if possible, can you help by creating an issue?

**Comment by escopecz** [comment]
@abhisekmazumdar I could find https://github.com/mautic/mautic/issues/14174

**Comment by andersonjeccel** [comment]
the specific failing e2e is strange, other e2e tests use the same function and path but are passing

image

File: app/bundles/CoreBundle/Assets/js/1a.content.js [hunk 1]
```diff
@@ -1469,6 +1469,65 @@ Mautic.activateLiveSearch = function (el, searchStrVar, liveCacheVar) {
     }
 };
 
+/**
+ * Filters a list based on select value
+ *
+ * @param el
+ */
+Mautic.activateListFilterSelect = function(el) {
```
Comment (escopecz) [inline app/bundles/CoreBundle/Assets/js/1a.content.js]: Where is this fn being used? I can't find a place.

Comment (andersonjeccel) [reply app/bundles/CoreBundle/Assets/js/1a.content.js]: within Mautic.onPageLoad, line 342

image

Comment (andersonjeccel) [reply app/bundles/CoreBundle/Assets/js/1a.content.js]: let me check something, wait a minute before merging

Comment (andersonjeccel) [reply app/bundles/CoreBundle/Assets/js/1a.content.js]: this function was refactored into another in the previous merged PR

rebased this branch, the function ceased existing

also tested now using the UI, all good

**Comment by andersonjeccel** [comment]
hm, my push didn't change anything related to the e2e test...

**Comment by abhisekmazumdar** [comment]
Yes, I can see that it's the same e2e test issue that I saw previously: `ContactManagementCest.php:importCSV`

**Pull request merged**